src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int trashould;
+    private int size;
+    private Node<K, V>[] table;
+
+    {
+        table = new Node[DEFAULT_CAPACITY];
+        trashould = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
+
+    private class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        private Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == trashould) {
+            resize();
+        }
+        Node<K, V> newNode = new Node(key, value, null);
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K,V> previousNode = table[index];
+            while (previousNode != null) {
+                if (previousNode.key == key
+                        || (previousNode.key != null && previousNode.key.equals(key))) {
+                    previousNode.value = value;
+                    return;
+                }
+                if (previousNode.next == null) {
+                    previousNode.next = newNode;
+                    size++;
+                    return;
```

Comment:
if u replace the `return` by the `break` u will be able to remove `size++` and leave only those one at the end of the method